src/widgets/modals/VaccinationEvent.js

```diff
@@ -84,17 +84,17 @@ export const VaccinationEventComponent = ({
   const [isDeleteModalOpen, toggleDeleteModal] = useToggle(false);
   const [vaccinator, setVaccinator] = useState(transactionBatch?.medicineAdministrator);
   const [vaccine, setVaccine] = useState(
-    vaccines.filter(item => item.id === transactionBatch?.itemId)
+    vaccines.filter(item => item.id === transactionBatch?.itemId)[0]
   );
   const vaccineDropDownValues = vaccines.map(({ code, name }) => `${code}: ${name}`);
 
   const [{ updatedPcdForm, isPCDValid }, setPCDForm] = useState({
     updatedPcdForm: null,
-    isPCDValid: false,
+    isPCDValid: true,
```

Comment:
It might not be necessary for the `isPCDValid` to set it to `true` by default but it is necessary for `isSupplementalDataValid` so I just wanted to make it consistent for both of them. 
Previously, `isPCDValid` was 'false' by default, but JSONForm component (where `isPCDValid` is actually being evaluated based on data being valid) was rerendered and `isPCDValid` was set to the correct value (which is `true` at the beginning of editing vaccination event always, invalid data cannot be saved in the UI). However, the JSONForm for `isSupplementalDataValid` wasn't rerendered so it wasn't evaluated and it was always `false` at the beginning even if the data is valid.
Sorry if my explanation is confusing, hopefully it made sense.